engine/crates/engine/src/context/ext.rs

```diff
@@ -64,9 +63,8 @@ pub trait ContextExt<'a>: Context<'a> {
     }
 
     fn trace_id(&self) -> String {
-        self.data::<Arc<DynamoDBBatchersData>>()
-            .map(|x| x.ctx.trace_id.clone())
-            .ok()
+        self.data::<Box<TraceId>>()
```

Comment:
Should this just be `TraceId` instead of `Box<TraceId>`?  Looks like you're not putting it in a box when you add it to the context elsewhere...

Reply:
You're right, fixed!